### ISSUE TITLE
Add sharing mechanism to synchronize on viewing the same Graph view

### ISSUE THREAD

**Issue by spockz** [kiali/kiali#3727]
**Is your feature request related to a problem? Please describe.**
When concurrently looking at the Graph View to troubleshoot issues we frequently had a different view to different display settings and/or different time views.

**Describe the solution you'd like**
Provide a share button that will provide a way to sync the view panel, either through sharing a picture, creating a link that includes the view panel time range and display options. What may be awesome might be something like a live shared graph that we can look at together and share highlights, but this might be overkill.

**Describe alternatives you've considered**
We created screenshots and posted them in teams or shared screens but they are a bit out of the normal flow.

**Additional context**
None

**Comment by israel-hdez** [comment]
Most Kiali pages are bookmarkable.
Have you tried to, simply, copy the link when you have all desired options set? Probably, it may be carrying some options you need. If not, mention what options are not being set in the URL.

I also need some clarification, as I don't understand what you mean by "concurrently looking at the Graph". Do you want to share the same view with colleagues? Or just have the same view on two monitors?...

Sounds like the Graph Replay feature may help here.... See: https://medium.com/kialiproject/kiali-sprint-34-overview-86027b13b27d#d6e6

**Comment by jshaughn** [comment]
@israel-hdez @spockz I think maybe what is suggested above is sort of what is being asked for.  If I am looking at an interesting graph, and I want to capture or share that exact graph [1], a "Capture" or "Share" button could be nice.  I think what it would be is what Edgar mentioned above, a URL that contained all of the options necessary to recreate that graph.  It would mean ensuring that we supported all Display options as URL params (today we support a subset), and that we could capture the time in some way.  That time capture is interesting, it may mean we'd need to be able to bookmark a replay range.

[1] with the understanding that istio config is always current, only the telemetry may be in the past

**Comment by spockz** [comment]
Indeed, the state/configuration of the graph being present in the url is useful for sharing. Having a button to quickly share it is a nice to have imho as cmd+l and then cmd+c also copies the url in the pastebuffer. 

Having a shared live view of the graph is maybe better handled with just sharing the screen.

**Comment by jshaughn** [comment]
+1, a shared live view is basically a video conference :)  A screenshot is useful for static images.  But there may still be a value add for this issue, by fully supporting a URL that can recreate the on-screen graph in a short graph replay (say 1m, maybe 5m).  It would be more robust than a screenshot and easier than doing a recording.  Not sure, trying to decide if there is something worth adding.

**Comment by spockz** [comment]
We regularly use the share functionality in Grafana which makes a copy of the data for that time window. This could be the analogue to that. Replay what went wrong at a crucial moment that caused the production incident. (Especially if it includes "events" like deployments and istio config changes.)

**Comment by lucasponce** [comment]
I'm reviewing this issue in the context to decide if it's still valid or to discard it.
Today, Kiali offers a powerful "Replay" feature that could be adjusted to recreate a given interval of time.
![image](https://user-images.githubusercontent.com/1662329/128699584-828ef93d-4ed4-42f9-a286-418738d6cd6a.png)

Perhaps can we detail what would be good to add to this feature to address the feature request ?

**Comment by spockz** [comment]
Hi @lucasponce,

I think the sharing feature is already quite good. The only thing that would be nice to have is to implement some synchronisation mechanism so people can look at the same picture while debugging without having to screen share. This would maybe require some web sockets/ssd to sync like reveal.js does.
